src/main/java/study/StringCalculator.java

```diff
@@ -0,0 +1,72 @@
+package study;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringCalculator {
+
+    public static final String DELIMETER = ",|:";
+    public static final String REGEX = "//(.)\n(.*)";
+
+    public static int  splitAndSum(String text) {
+        if (isBlank(text)) {
+            return 0;
+        }
+
+        Matcher m = Pattern.compile(REGEX).matcher(text);
+
+        if (isCustomDelimeter(m)) {
+            String customDelimiter = m.group(1);
+            String[] values = m.group(2).split(customDelimiter);
+
+            return sum(values);
+        }
+
+        String[] values = text.split(DELIMETER);
+
+        return sum(values);
+    }
```

Comment:
하나의 메서드에서 여러가지 작업을 수행 중인데요. 이에 추상화 수준이 좀 다른 거 같습니다. 취향 차이긴 하지만, if 조건 분기를 어떻게 숨겨볼까 고민해보다 보면 여러 형태로 코드를 작성해 볼 수 있어요~ 이번 과정에서 한번 도전해보시는 것도 좋겠어요~

```suggestion
    public static int splitAndSum(String text) {
        if (isBlank(text)) {
            return 0;
        }

        return sum(split(text));
    }

    public static String[] split(String text) {
        return customDelimiterMatcher(text)
                .map(StringCalculator::split)
                .orElseGet(() -> text.split(DELIMETER));
    }
```